### DISCUSSION TITLE
how can i close modal which I wanna control with state

### DISCUSSION THREAD

**Discussion by dkumza** [saadeghi/daisyui#3120] [Q&A] [answered]
I'm a bit confused, I show modal like this: 
```jsx
  const handleShowModal = () => {
    const modal = document.getElementById('exp_modal');
    modal && modal.showModal();
    dispatch(openModal());
  };
```
and my modal looks like this: 
```jsx
 <>
      <dialog id='exp_modal' className='modal'>
        <div className='modal-box flex justify-center '>
          <ExpensesPage />
        </div>
        <form method='dialog' className='modal-backdrop'>
          <button className='' onClick={() => dispatch(closeModal())}>
            close
          </button>
        </form>
      </dialog>
    </>
```
I wanna close modal not only by "close" button, but somewhere in my app in different places.
and I do have current state. Which makes a bit buggy, even i have current state i can't control modal.
How do you guys control it? I wanna do this is few other components, by changing state...

**Comment by saadeghi** [comment, marked as answer]
`.close()` method.

HTML `<dialog>` docs:  
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog
